### PR TITLE
style show qualification page based on mockup

### LINKED ISSUE #231: Style the show-qualification page according to mockups
See avocode page 'Qualification - Information review'

### LINKED ISSUE #306: render interests, jobs and important dates to the qualification view

### PR DESCRIPTION
**Pull request by boney-bun**
fix #231
fix #306

### CODE REVIEW

**Comment by boney-bun** [comment]
the GIF:
ford3#248

**Comment by pierrealixt** [comment]
@boney-bun 
What about `3. Requirements`, `4.Interests and Jobs` and `5. Important Dates`?

**Comment by boney-bun** [comment]
> What about 3. Requirements, 4.Interests and Jobs and 5. Important Dates?

seems to me that these sections haven't developed yet.
however, i apply the same style as in *Details* and *Duration and Fees* sections.

**Comment by pierrealixt** [comment]
@boney-bun 
On your screenshot, there are some requirements data.

**Comment by boney-bun** [comment]
yeah... i added some dummy requiement data to check the style, but it doesn't populate properly.
i don't understand the busines logic either for those sections.
that's why i start to think that those sections are left to be developed later.

**Comment by Christiaanvdm** [comment]
It needs to look like this - Ignore the percentages + their bars for now though.
Qualification Information_Review

**Comment by boney-bun** [comment]
@Christiaanvdm i just created a new issue (#287) regarding my previous comment

**Comment by boney-bun** [comment]
PIA:
ford3#QualificationShow

questions: 
- what's the difference between `back` and `go to campus overview` buttons?
i think both of these buttons will refer to the same page.
- how to parse saqa.saqaqualificationlevel ?

**Comment by pierrealixt** [comment]
Yes you can safely remote the `back` button.
What's the issue with `saqa.saqaqualificationlevel`?

**Comment by boney-bun** [comment]
> What's the issue with saqa.saqaqualificationlevel?

it shows `saqaqualificationlevel.LEVEL1` in `required entrance qualification` (refers to the gif).
i'm not sure if this is the correct output.

**Comment by pierrealixt** [comment]
@boney-bun have you checked the file `ford3/enums/saqa_qualification_level.py`?

**Comment by boney-bun** [comment]
@pierrealixt the nqf level problem is solved using templatetags:
ford3#248_rev

**Comment by boney-bun** [comment]
the latest changes:
ford3#248_rev

File: django_project/ford3/models/qualification.py [hunk 1]
```diff
@@ -129,6 +129,16 @@ def interest_id_list(self) -> List[int]:
             result.append(each_item['id'])
         return result
 
+    @property
+    def interest_name_list(self) -> List[int]:
+        result = []
+        interest_query = Interest.objects.filter(
```
Comment (pierrealixt) [inline django_project/ford3/models/qualification.py:142]: see list comprehension in python.
`result [each_item['name'] for each_item in list(interest_query)`

File: django_project/ford3/templates/qualification.html [hunk 2]
```diff
@@ -8,77 +9,175 @@
...
 <div class='container'>
     <div class='row'>
-        <div class='col-md-12'>
-            <h3>{{ qualification.campus }} {{ qualification }}</h3>
-
-            <div class='row'>
-                <div class='col-md-10'>
-                    <h4>1. Details</h4>
+        <div class="col-md-12">
+            <h3 class="mb-4 noplr">{{ qualification.campus }}: {{ qualification }}</h3>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: use `px-0`

File: django_project/ford3/templates/qualification.html [hunk 3]
```diff
@@ -8,77 +9,175 @@
...
+            <div class='row pt-3 nomlr'>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: use `mx-0`

File: django_project/ford3/templates/qualification.html [hunk 4]
```diff
@@ -8,77 +9,175 @@
...
+                <div class='col-md-10 edu-underline noplr'>
+                    <h4 class="mt-3">1. Details</h4>
                 </div>
                 <div class='col-md-2'>
                     <a class="edu-button edu-button-orange" href="{% url 'edit-qualification' provider_id=qualification.campus.provider.id campus_id=qualification.campus.id qualification_id=qualification.id %}">Edit</a>
                 </div>
             </div>
-            <div class='row'>
+            <div class='row pt-3 pb-3'>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: `py-3`

File: django_project/ford3/templates/qualification.html [hunk 5]
```diff
@@ -8,77 +9,175 @@
...
                 <div class='col-md-12'>
-                    <b>Short Description of this Qualification</b>
-                    <p>{{ qualification.short_description }}</p>
+                    <b>Short description of this qualification:</b>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: Short description

File: django_project/ford3/templates/qualification.html [hunk 6]
```diff
@@ -8,77 +9,175 @@
...
+                    <p class="pt-3">{{ qualification.short_description }}</p>
 
-                    <b>Long Description of this Qualification</b>
-                    <p>{{ qualification.long_description }}</p>
+                    <b>Long description of this qualification:</b>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: Long description

File: django_project/ford3/templates/qualification.html [hunk 7]
```diff
@@ -8,77 +9,175 @@
...
+                    <p class="pt-3 pb-3">{{ qualification.long_description }}</p>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: py-3

File: django_project/ford3/templates/qualification.html [hunk 8]
```diff
@@ -8,77 +9,175 @@
...
+                            <div class="col-md-6">
+                                <b>Subject {{ forloop.counter }}: </b>{{ subject.name }} {{ subject.minimum_score }}%
+                            </div>
+                        {% endfor %}
+                        </div>
+                    {% endfor %}
+                </div>
+            </div>
+            <div class="row pt-3 pb-3 nomlr">
+                <div class='col-md-10 edu-underline noplr'>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: please edit from
```
.edu-underline {
   border-bottom: 1px solid #d5d5d5
}
```

to 
```
.edu-underline {
   border-bottom: 1px solid var(--c-grey-l)
}
```

File: django_project/ford3/templates/qualification.html [hunk 9]
```diff
@@ -8,77 +9,175 @@
...
+                </div>
+                <div class='col-md-2'>
+                    <a class="edu-button edu-button-orange" href="#">Edit</a>
+                </div>
+            </div>
+            <div class='row pt-3 pb-3'>
                 <div class='col-md-12'>
+                    <div class="row pb-3">
+                        <div class="col-md-6">
+                            <b>Choose three interests associated to this qualification:</b>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: Interests

File: django_project/ford3/templates/qualification.html [hunk 10]
```diff
@@ -8,77 +9,175 @@
...
+                            <br />
+                            {% for  interest in qualification.interest_name_list %}
+                                {{  interest }}
+                                {% if forloop.counter < qualification.interest_name_list|length %}, {% endif  %}
+                            {% endfor %}
+                        </div>
+                        <div class="col-md-6">
+                            <b>Choose up to five occupations that this qualification could prepare you for:</b>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: Occupations

File: django_project/ford3/templates/qualification.html [hunk 11]
```diff
@@ -8,77 +9,175 @@
...
+                            <br />
+                            {% for occupation in qualification.occupation_name_list %}
+                                {{  occupation }}
+                                {% if forloop.counter < qualification.occupation_name_list|length %}, {% endif  %}
+                            {% endfor %}
+                        </div>
+                    </div>
+                    <div class="row pb-3">
+                        <div class="col-md-6">
+                            <b>Does the qualification prepare for a critical skills occupation:</b>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: we could use a green check similar to  ☑️if the value is True
and ❌if the value is False
or use checkbox input in read-only

File: django_project/ford3/templates/qualification.html [hunk 12]
```diff
@@ -8,77 +9,175 @@
...
+                            {{ qualification.high_demand_occupation|yesno:"YES,NO" }}
+                        </div>
+                    </div>
 
                 </div>
             </div>
+
+            <div class="row pt-3 pb-3 nomlr">
+                <div class='col-md-10 edu-underline noplr'>
+                    <h4 class="mt-3">5. Important Dates</h4>
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: Important dates

Comment (pierrealixt) [reply django_project/ford3/templates/qualification.html]: Events

File: django_project/ford3/templates/qualification.html [hunk 13]
```diff
@@ -8,77 +9,175 @@
...
+                </div>
+                <div class='col-md-2'>
+                    <a class="edu-button edu-button-orange" href="#">Edit</a>
+                </div>
+            </div>
+            <div class='row pt-3 pb-3'>
+                <div class='col-md-6'>
+                    <b>Application period start: </b>{{ qualification.qualification_events_list.0.date_start }}
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: There can be more than one event.
Each event has a name

File: django_project/ford3/templates/qualification.html [hunk 14]
```diff
@@ -8,77 +9,175 @@
...
+                </div>
+                <div class='col-md-6'>
+                    <b>Application period end: </b>{{ qualification.qualification_events_list.0.date_end }}
+                </div>
+            </div>
+            <div class="row mb-5">
+                <div class="col-md-9">
+                    {# spacer #}
+                </div>
+                <div class="col-md-3">
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: add the class `col-md-offset-9`

Comment (pierrealixt) [reply django_project/ford3/templates/qualification.html]: or you do
```
<div class='col-md-12 text-right'>
<a href=''></a>
</div>
```

Comment (pierrealixt) [reply django_project/ford3/templates/qualification.html]: center the button instead
```
<div class='row'>
<div class='col-md-2 mx-auto'>
your button
</div>
</div>
```

**Comment by pierrealixt** [comment]
@boney-bun 

Screen Shot 2019-05-13 at 09 40 38
Have you changed something here?


When the qualification is fresh, this is what it looks like:
Screen Shot 2019-05-13 at 09 41 24
Maybe we could have another text instead of None. What do you think?

File: django_project/ford3/templates/qualification.html [hunk 15]
```diff
@@ -12,75 +13,181 @@
...
+                        <div class="col-md-6">
+                            <b>Does the qualification prepare for a green occupation</b>
+                            {% if qualification.green_occupation  %}
+                                &#9989
+                            {% else %}
+                                &#10060
+                            {% endif %}
+                        </div>
+                    </div>
+                    <div class="row pb-3">
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html:145]: it's weird to re-create a new `.row`.
And at the end, it doesn't look good
do something like
```
<p>
<b>Does the qualification prepare for a critical skills occupation</b>
<br />
...
</p>
```

File: django_project/ford3/templates/qualification.html [hunk 16]
```diff
@@ -12,75 +13,181 @@
...
+                        <div class="col-md-6">
+                            <b>Does the qualification prepare for a high demand occupation</b>
+                            {% if qualification.high_demand_occupation %}
+                                &#9989
```
Comment (pierrealixt) [inline django_project/ford3/templates/qualification.html]: code repetition here, you could use a template tag?

**Comment by boney-bun** [comment]
@pierrealixt sorry, it was because `noplr` has been removed. i've changed using `px-0`. so there will be no more bug in sidebar.
i use templatetags for changing None value in the string and number

if the text is None:
Screen Shot 2019-05-13 at 4 42 41 PM

Note: i'm not sure about the `NQF Level`. it can be left empty. what should we replace in case of empty? `Not required`?

File: django_project/ford3/templates/qualification.html (hunk 15 above)
Comment (boney-bun) [reply django_project/ford3/templates/qualification.html:145]: using `<p>` will result:

Screen Shot 2019-05-13 at 3 56 33 PM

the use of `.row` is to make sure that they are in a row.
the `interests` and `occupations` can be of different height.
do you still want to use `<p>` instead?

Comment (pierrealixt) [reply django_project/ford3/templates/qualification.html:145]: I dont know Boney, does it look good to you?
Maybe I will get rid of the loooong question `Does the qualification....`
Just 'critical skills', ...

**Pull request merged**